eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -295,17 +295,38 @@ else if (inside.size() > 32 && !(inside instanceof SetIterable))
     /**
      * Returns true if all elements in source are contained in this collection.
      *
-     * @since 1.0
+     * @since 11.1
      */
-    boolean containsAllIterable(Iterable<?> source);
+    default boolean containsAllIterable(Iterable<?> source)
+    {
+        if (source instanceof RichIterable)
+        {
+            RichIterable<?> outside = this;
+            RichIterable<?> inside = (RichIterable<?>) source;
+            if (this.size() < inside.size())
+            {
+                return false;
+            }
+            if (outside.size() > 32 && !(outside instanceof SetIterable))
+            {
+                outside = outside.toSet();
+            }
```

Comment:
How did you pick 32?

Reply:
To make the implementation consistent with `RichIterable#containsNoneIterable`.